src/plugin/areadata/areadataplugin.js

```diff
@@ -0,0 +1,120 @@
+goog.module('plugin.areadata.AreaDataPlugin');
+goog.module.declareLegacyNamespace();
+
+
+const AbstractPlugin = goog.require('os.plugin.AbstractPlugin');
+const registry = goog.require('os.ogc.registry');
+const services = goog.require('os.ogc.services');
+const OGCService = goog.require('os.ogc.OGCService');
+const {AreaImportType} = goog.require('plugin.areadata');
+
+const OSSettings = goog.require('os.config.Settings');
+
+const Settings = OSSettings.getInstance();
+
+
+/**
+ * @type {string}
+ * @const
+ */
+const ID = 'areadata';
+
+/**
+ * @type {string}
+ * @const
+ */
+const KEYS_SETTING = 'areadata.keys';
+
+/**
+ * Maps a non-minimized os.ui.menu.MenuItemOptions object, e.g. from settings.json, onto the minimized version
+ *
+ * WARNING: To avoid turning a common object into an extern, only the primitive properties defined as of 20200812
+ * are mapped to a new os.ui.menu.MenuItemOptions object. Please add new mappings as needed.
+ *
+ * @param {Object.<string, *>|undefined|null} config
+ * @return {os.ui.menu.MenuItemOptions|undefined}
+ */
+const minify_ = function(config) {
+  if (!config) return undefined;
+
+  return /** @type {os.ui.menu.MenuItemOptions} */ ({
+    eventType: config['eventType'],
+    label: config['label'],
+    metricKey: config['metricKey'],
+    visible: config['visible'],
+    enabled: config['enabled'],
+    selected: config['selected'],
+    icons: config['icons'],
+    tooltip: config['tooltip'],
+    shortcut: config['shortcut'],
+    sort: config['sort']
+  });
+};
+
+/**
+ *
+ */
+class AreaDataPlugin extends AbstractPlugin {
+  /**
+   *
+   */
+  constructor() {
+    super();
+
+    this.id = ID;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  init() {
+    // Read the settings.json and initialize the Area importer(s)...
+    // NOTE: the "arraylike" object returned by Settings is not iterable; pass it through an Array.from()
```

Comment:
This... should not be the case. Settings returns standard strings, objects, and arrays that should all be iterable in the normal ways. They aren't like `NodeList`s or other weird `ArrayLike` structures.

Reply:
This one was weird -- it worked just fine in the full code, but returned an "almost an array" array in the compiled code. So the following .forEach(() => {}) function didn't work even though .length worked (a property that ONLY exists on Arrays (and Strings, which are Arrays).